StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1203SA1214CodeFixProvider.cs

```diff
@@ -0,0 +1,102 @@
+﻿namespace StyleCop.Analyzers.OrderingRules
+{
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using StyleCop.Analyzers.Helpers;
+
+    /// <summary>
+    /// Implements code fixes for <see cref="SA1203ConstantsMustAppearBeforeFields"/>.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(SA1203SA1214CodeFixProvider))]
+    [Shared]
+    public class SA1203SA1214CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(SA1203ConstantsMustAppearBeforeFields.DiagnosticId, SA1214StaticReadonlyElementsMustAppearBeforeStaticNonReadonlyElements.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return CustomFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (Diagnostic diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
+            {
+                context.RegisterCodeFix(CodeAction.Create(OrderingResources.SA1203CodeFix, token => GetTransformedDocumentAsync(context.Document, diagnostic, token), equivalenceKey: nameof(SA1203SA1214CodeFixProvider)), diagnostic);
+            }
+
+            return SpecializedTasks.CompletedTask;
+        }
+
+        private static async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var fieldDeclaration = syntaxRoot.FindNode(diagnostic.Location.SourceSpan).FirstAncestorOrSelf<FieldDeclarationSyntax>();
+            var typeDeclarationNode = fieldDeclaration.FirstAncestorOrSelf<TypeDeclarationSyntax>();
+            if (typeDeclarationNode == null)
+            {
+                return document;
+            }
+
+            var allFields = GetOrderHelperForFields(typeDeclarationNode);
+            var fieldToMove = new MemberOrderHelper(fieldDeclaration);
+            for (var i = 0; i < allFields.Count; i++)
+            {
+                if (allFields[i].ModifierPriority < fieldToMove.ModifierPriority)
```

Comment:
:bulb: Get the index of `fieldDeclaration` in `typeDeclarationNode.Members` and iterate backwards over members. Instead of enumerating the entire collection in `GetOrderHelperForFields`.